pppd/auth.c

```diff
@@ -2666,6 +2670,14 @@ get_eaptls_secret(int unit, char *client, char *server,
 			strlcpy( capath, ca_path, MAXWORDLEN );
 		strlcpy( pkfile, privkey_file, MAXWORDLEN );
 	}
+    else if (!am_server && pkcs12_file)
```

Comment:
The pkcs12 array gets initialized on this path, but not if this path isn't taken. Since the caller is passing in an uninitialized stack array, and you subsequently test pcks12[0], you need either to initialize pcks12 on all paths through this function, or else initialize it in all callers (the former is probably easier).